src/modules/cpu/kernel/fog.hpp

```diff
@@ -0,0 +1,1524 @@
+        {
+            Rpp8u *srcPtrRow, *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            Rpp32f *fogAlphaMaskPtrRow, *fogIntensityMaskPtrRow;
+            srcPtrRow = srcPtrChannel;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+            fogAlphaMaskPtrRow = fogAlphaMaskPtr;
+            fogIntensityMaskPtrRow = fogIntensityMaskPtr;
+
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTemp, *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                Rpp32f *fogAlphaMaskPtrTemp, *fogIntensityMaskPtrTemp;
+                srcPtrTemp = srcPtrRow;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+                fogAlphaMaskPtrTemp = fogAlphaMaskPtrRow;
+                fogIntensityMaskPtrTemp = fogIntensityMaskPtrRow;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                {
+
+                    __m256 pfogAlphaMask[2], pfogIntensityMask[2], p[6];
+                    rpp_simd_load(rpp_load16_f32_to_f32_avx, fogAlphaMaskPtrTemp, pfogAlphaMask);    // simd loads
+                    rpp_simd_load(rpp_load16_f32_to_f32_avx, fogIntensityMaskPtrTemp, pfogIntensityMask);    // simd loads
+                    rpp_simd_load(rpp_load48_u8pkd3_to_f32pln3_avx, srcPtrTemp, p);    // simd loads
+                    compute_fog_48_host(p, pfogAlphaMask, pfogIntensityMask);    // fog adjustment
+                    rpp_simd_store(rpp_store48_f32pln3_to_u8pln3_avx, dstPtrTempR, dstPtrTempG, dstPtrTempB, p);    // simd stores
+
+                    srcPtrTemp += vectorIncrement;
+                    dstPtrTempR += vectorIncrementPerChannel;
+                    dstPtrTempG += vectorIncrementPerChannel;
+                    dstPtrTempB += vectorIncrementPerChannel;
+                    fogAlphaMaskPtrTemp += vectorIncrementPerChannel;
+                    fogIntensityMaskPtrTemp += vectorIncrementPerChannel;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount += 3)
+                {
+
+                    
+                    *dstPtrTempR = (Rpp8u) RPPPIXELCHECK(std::nearbyintf(((Rpp32f) srcPtrTemp[0]) * ( 1 - *fogAlphaMaskPtrTemp) + (*fogIntensityMaskPtrTemp * (*fogAlphaMaskPtrTemp))));
+                    *dstPtrTempG = (Rpp8u) RPPPIXELCHECK(std::nearbyintf(((Rpp32f) srcPtrTemp[1]) * ( 1 - *fogAlphaMaskPtrTemp) + (*fogIntensityMaskPtrTemp * (*fogAlphaMaskPtrTemp))));
+                    *dstPtrTempB = (Rpp8u) RPPPIXELCHECK(std::nearbyintf(((Rpp32f) srcPtrTemp[2]) * ( 1 - *fogAlphaMaskPtrTemp) + (*fogIntensityMaskPtrTemp * (*fogAlphaMaskPtrTemp))));
+
```

Comment:
Since the alphaMask and intensity mask values are going to be same for each channel
You can store them in 2 seperate variables and re use them, instead of doing the compute 3 times

Rpp32f alphaFactor = ( 1 - *fogAlphaMaskPtrTemp)
Rpp32f intensityFactor = (*fogIntensityMaskPtrTemp * (*fogAlphaMaskPtrTemp));

reuse this 2 values for output calculation